libraries/shared/storage/diff_repository.go

```diff
@@ -62,37 +62,11 @@ func (repository diffRepository) CreateBackFilledStorageValue(rawDiff types.RawD
 	return err
 }
 
-func (repository diffRepository) GetNewDiffs(diffs chan types.PersistedDiff, errs chan error, done chan bool) {
-	rows, queryErr := repository.db.Queryx(`SELECT * FROM public.storage_diff WHERE checked IS false`)
-	if queryErr != nil {
-		logrus.Errorf("error getting unchecked storage diffs: %s", queryErr.Error())
-		if rows != nil {
-			closeErr := rows.Close()
-			if closeErr != nil {
-				logrus.Errorf("error closing rows: %s", closeErr.Error())
-			}
-		}
-		errs <- queryErr
-		return
-	}
-
-	if rows != nil {
-		for rows.Next() {
-			var diff types.PersistedDiff
-			scanErr := rows.StructScan(&diff)
-			if scanErr != nil {
-				logrus.Errorf("error scanning diff: %s", scanErr.Error())
-				closeErr := rows.Close()
-				if closeErr != nil {
-					logrus.Errorf("error closing rows: %s", closeErr.Error())
-				}
-				errs <- scanErr
-			}
-			diffs <- diff
-		}
-	}
-
-	done <- true
+func (repository diffRepository) GetNewDiffs(minID, limit int) ([]types.PersistedDiff, error) {
+	var result []types.PersistedDiff
+	query := fmt.Sprintf("SELECT * FROM public.storage_diff WHERE checked IS false and id > %d ORDER BY id ASC LIMIT %d", minID, limit)
```

Comment:
Feels like it might be more intuitive to filter by block height rather than ID, but I shied away from that since:
- block numbers aren't unique, so it's theoretically possible we could fail to parse more than 500 diffs from a single block on the first pass through
- vague inklings that querying by id might be more efficient